src/families/cosmos/react.js

```diff
@@ -14,3 +24,143 @@ export function useCosmosPreloadData() {
   }, []);
   return state;
 }
+
+export function useCosmosFormattedDelegations(
+  account: Account,
+  option?: FormatOption
+): CosmosFormattedDelegation[] {
+  const { validators } = useCosmosPreloadData();
+  const delegations = account.cosmosResources?.delegations;
+  invariant(delegations, "cosmos: delegations is required");
+
+  const formattedDelegations = useMemo(
+    () => formatDelegations(delegations, validators),
```

Comment:
shouldn't the switch be INSIDE this? the value of useMemo could only be useful if we move the switch **inside** the function too and that we also accept an `option` as useMemo dependency.